tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/aux/RecordAndVariantOps.scala

```diff
@@ -292,15 +292,15 @@ class RecordAndVariantOps(rewriter: SymbStateRewriter) {
     val fieldTypes = getFieldTypes(recordCell)
     expectFieldName(recordCell, fieldTypes, fieldName)
     val index = fieldTypes.keySet.toList.indexOf(fieldName)
-    val elems = state.arena.getHas(recordCell)
+    val elems = state.arena.getHasPtr(recordCell)
     assert(0 <= index && index < elems.length)
 
     var nextState = state.updateArena(_.appendCell(recordCell.cellType.toTlaType1))
     val newRecord = nextState.arena.topCell
     // add the fields in the order of their names, update by name
-    for ((name, oldCell) <- fieldTypes.keySet.zip(elems)) {
-      val updatedCell = if (name == fieldName) newCell else oldCell
-      nextState = nextState.updateArena(_.appendHasNoSmt(newRecord, SmtConstElemPtr(updatedCell)))
+    for ((name, oldPtr) <- fieldTypes.keySet.toSeq.zip(elems)) {
```

Comment:
Any particular reason why `getHasPtr`/`getHas` return `Seq`s rather than `Set`s?

Reply:
Yes, in arenas, has-pointers are ordered. E.g. for seqs, the cell order in has matches the element order in the seq.

Reply:
Ah makes sense, ty!